src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -238,8 +255,7 @@ DirectStepI::revImp (
 
     bool srcRedeems;
     IOUAmount maxSrcToDst;
-    std::tie (maxSrcToDst, srcRedeems) =
-        maxFlow (sb, src_, dst_, currency_);
+    std::tie (maxSrcToDst, srcRedeems) = maxFlow (sb, out);
```

Comment:
If the intent is to ignore trust lines for offer crossing, this isn't quite right. `maxSrcToDst` is compared to `srcToDst`, not `out`, so it would be better to use `srcToDst` as the requested amount. Another option is to keep `maxFlow` as is and take the logic that ignores trust lines and make callers decide if they want to ignore the trust line.